ale/base/type_sensor.py

```diff
@@ -362,3 +368,126 @@ def line_times(self):
+                self._camera_model_group['O'] = np.array(camera_model_group["MODEL_COMPONENT_5"])
+                self._camera_model_group['R'] = np.array(camera_model_group["MODEL_COMPONENT_6"])
+        return self._camera_model_group
+
+    @property
+    def cahvor_rotation_matrix(self):
+        """
+        Computes the cahvor rotation matrix for the instrument to Rover frame
+
+        Returns
+        -------
+        : array
+          Rotation Matrix as a 2D numpy array
+        """
+        if not hasattr(self, "_cahvor_rotation_matrix"):
+            h_c = np.dot(self.cahvor_camera_params['A'], self.cahvor_camera_params['H'])
+            h_s = np.linalg.norm(np.cross(self.cahvor_camera_params['A'], self.cahvor_camera_params['H']))
+            v_c = np.dot(self.cahvor_camera_params['A'], self.cahvor_camera_params['V'])
+            v_s = np.linalg.norm(np.cross(self.cahvor_camera_params['A'], self.cahvor_camera_params['V']))
+            H_prime = (self.cahvor_camera_params['H'] - h_c * self.cahvor_camera_params['A'])/h_s
+            V_prime = (self.cahvor_camera_params['V'] - v_c * self.cahvor_camera_params['A'])/v_s
+            r_matrix = np.array([H_prime, -V_prime, -self.cahvor_camera_params['A']])
+
+            phi = math.asin(r_matrix[2][0])
+            w = - math.asin(r_matrix[2][1] / math.cos(phi))
+            k = math.acos(r_matrix[0][0] / math.cos(phi))
+
+            w = math.degrees(w)
+            phi = math.degrees(phi)
+            k = math.degrees(k)
+
+            # Rotational Matrix M generation
+            self._cahvor_rotation_matrix = np.zeros((3, 3))
+            self._cahvor_rotation_matrix[0, 0] = math.cos(phi) * math.cos(k)
+            self._cahvor_rotation_matrix[0, 1] = math.sin(w) * math.sin(phi) * math.cos(k) + \
+                math.cos(w) * math.sin(k)
+            self._cahvor_rotation_matrix[0, 2] = - math.cos(w) * math.sin(phi) * math.cos(k) + \
+                math.sin(w) * math.sin(k)
+            self._cahvor_rotation_matrix[1, 0] = - math.cos(phi) * math.sin(k)
+            self._cahvor_rotation_matrix[1, 1] = - math.sin(w) * math.sin(phi) * math.sin(k) + \
+                math.cos(w) * math.cos(k)
+            self._cahvor_rotation_matrix[1, 2] = math.cos(w) * math.sin(phi) * math.sin(k) + \
+                math.sin(w) * math.cos(k)
+            self._cahvor_rotation_matrix[2, 0] = math.sin(phi)
+            self._cahvor_rotation_matrix[2, 1] = - math.sin(w) * math.cos(phi)
+            self._cahvor_rotation_matrix[2, 2] = math.cos(w) * math.cos(phi)
+        return self._cahvor_rotation_matrix
+
+    @property
+    def frame_chain(self):
+        """
+        Returns a modified frame chain with the cahvor models extra rotation
+        added into the model
+
+        Returns
+        -------
+        : object
+          A networkx frame chain object
+        """
+        if not hasattr(self, '_frame_chain'):
+            nadir = self._props.get('nadir', False)
```

Comment:
Nadir should also not be supported for CAVHOR, because if the pointing is missing, it is an invalid CAVHOR model.